receiver/filelogreceiver/filelog_test.go

```diff
@@ -385,3 +385,78 @@ func (g *fileLogGenerator) Generate() []receivertest.UniqueIDAttrVal {
 	require.NoError(g.t, err)
 	return []receivertest.UniqueIDAttrVal{id}
 }
+
+// TestSymlinkedLogs tests that the filelog receiver reading from a single
+// file that's actually a symlink to another file, while the symlink target
+// is changed frequently, reads all the logs from all the files ever targeted
+// by that symlink.
+func TestSymlinkedLogs(t *testing.T) {
```

Comment:
Can we implement this test in the `fileconsumer` package? Generally I think it's preferable to localize the tests as much as possible so that the package which implements the functionality is fully tested.